src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -1381,6 +1381,20 @@
   (let [last-modified-milliseconds (.lastModified (io/file cacrl))]
        (time-coerce/from-long last-modified-milliseconds)))
 
+(schema/defn ^:always-validate update-crl :- CertificateRevocationList
+  [upstream-crl-pem :- InputStream
+   crl-path :- schema/Str]
+  (log/info (i18n/trs "Updating CRL at {0}" crl-path))
+  (let [upstream-crls (utils/pem->crls upstream-crl-pem)
+        our-crl (first (utils/pem->crls crl-path))
+        new-crl-chain (cons our-crl upstream-crls)]
+    ;; what verification do we need?
+    ;; do we need to confirm that a custom intermediate CA is being used?
+    ;; do we need to verify everything against the cert chain? i.e. confirm the order?
```

Comment:
I think doing this will answer all of the others, i.e. the verification will fail if any of those other things are true.